node-server/src/services/database.js

```diff
@@ -0,0 +1,122 @@
+const logger    = require('../../../common/utils/logger')
+const testData = require('../utils/test-data')
+const querier = require('../database/querier')
+const Dao = require('../database/dao')
+let dao = new Dao(querier)
+
+/**
+ * @typedef {import('../../../common/utils/types/datatypes).Message} Message
+ * @typedef {import('../../../common/utils/types/datatypes).Chat} Chat
+ */
+class DatabaseService {
+
+  /**
+   * Initiates database
+   */
+  initiateDatabase = async () => {
+    querier.initiateDatabase()
+  }
+
+  /**
+   * Opens connection and initiates tables
+   * @param {Querier} q
+   */
+  openDatabaseConnection = async (d = dao) => {
+    dao = d
```

Comment:
```suggestion
openDatabaseConnection = async (d = Dao(querier)) => {

```
How about this (if it works, I don't know)?

Reply:
If I understand the logic correctly this might cause crashes? That is, querier might be uninitialized when instantiating a new Dao object and thus db refers to null? Should querier be initialized in a constructor?